sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -277,6 +277,7 @@ object SQLConf {
       "Note: For structured streaming, this configuration cannot be changed between query " +
       "restarts from the same checkpoint location.")
     .intConf
+    .checkValue(_ > 0, "The value of spark.sql.shuffle.partitions must be positive")
```

Comment:
Just in case, how much positive could it be? `Int.MaxValue`?

Reply:
In theory, yeah. I'm not sure if we check every int parameter to verify it can successfully accept an int; that seems like it's tested by tests of the conf framework. Checking for -1, sure.